dependencies/src/main/java/com/google/cloud/tools/opensource/classpath/StaticLinkageChecker.java

```diff
@@ -182,6 +187,90 @@ private static void printStaticLinkageErrors(
+
+    // TODO(suztomo): implement validation for field, method and class references in the table
+    return reportBuilder.build();
+  }
+
+  /**
+   * Scans class files in the jar file and creates {@link SymbolTable} populated with class names
+   * defined within the jar and symbolic references to other classes outside the jar.
+   *
+   * @param jarFilePath absolute path to a jar file.
+   * @return Table containing symbol references and classes defined in the jar file.
+   */
+  static SymbolTable scanExternalSymbolTable(Path jarFilePath) {
+    SymbolTable.Builder symbolTableBuilder = SymbolTable.builder();
+    ImmutableSet.Builder<String> symbolTableClassNameBuilder = ImmutableSet.builder();
+    ImmutableSet.Builder<ClassSymbolReference> classSymbolReferences = ImmutableSet.builder();
+    ImmutableSet.Builder<MethodSymbolReference> methodSymbolReferences = ImmutableSet.builder();
+    ImmutableSet.Builder<FieldSymbolReference> fieldSymbolReferences = ImmutableSet.builder();
+
+    String pathToJar = jarFilePath.toString();
+    SyntheticRepository repository = SyntheticRepository.getInstance(new ClassPath(pathToJar));
+
+    try {
+      for (JavaClass javaClass : ClassDumper.topLevelJavaClassesInJar(jarFilePath, repository)) {
+        SymbolTable partialSymbolTable = ClassDumper.scanClassSymbolTable(javaClass);
+
+        classSymbolReferences.addAll(partialSymbolTable.getClassReferences());
+        methodSymbolReferences.addAll(partialSymbolTable.getMethodReferences());
+        fieldSymbolReferences.addAll(partialSymbolTable.getFieldReferences());
+        symbolTableClassNameBuilder.addAll(partialSymbolTable.getDefinedClassNames());
+      }
+      ImmutableSet<String> classesDefinedInJar = symbolTableClassNameBuilder.build();
+
+      ImmutableSet<ClassSymbolReference> externalClassReferences = classSymbolReferences.build()
+          .stream()
+          .filter(reference -> !classesDefinedInJar.contains(reference.getTargetClassName()))
+          .collect(ImmutableSet.toImmutableSet());
+      ImmutableSet<MethodSymbolReference> externalMethodReferences = methodSymbolReferences.build()
+          .stream()
+          .filter(reference -> !classesDefinedInJar.contains(reference.getTargetClassName()))
+          .collect(ImmutableSet.toImmutableSet());
+      ImmutableSet<FieldSymbolReference> externalFieldReferences = fieldSymbolReferences.build()
+          .stream()
+          .filter(reference -> !classesDefinedInJar.contains(reference.getTargetClassName()))
+          .collect(ImmutableSet.toImmutableSet());
+
+      return symbolTableBuilder
+          .setClassReferences(externalClassReferences)
+          .setMethodReferences(externalMethodReferences)
+          .setFieldReferences(externalFieldReferences)
+          .setDefinedClassNames(classesDefinedInJar)
+          .build();
+    } catch (ClassNotFoundException | IOException ex) {
+      throw new RuntimeException("Failed to scan jar file", ex);
+    }
+  }
+
+
```

Comment:
Delete extra blank line.

Can we use google-java-format for every PR? Specifically in the mode that formats only changed lines.